packages/eas-cli/src/project/remoteVersionSource.ts

```diff
@@ -98,3 +134,96 @@ export function getBuildVersionName(platform: Platform): string {
   }
 }
+
+export async function ensureAppVersionSourceIsSetAsync(
+  easJsonAccessor: EasJsonAccessor,
+  easJsonCliConfig: EasJson['cli'] | undefined,
+  nonInteractive: boolean
+): Promise<EasJson['cli'] | undefined> {
+  let selectOption, updateEasJson;
+  if (nonInteractive) {
+    Log.warn(
+      `The field "cli.appVersionSource" is not set, but it will be required in the future. ${learnMore(
+        'https://docs.expo.dev/build-reference/app-versions/'
+      )}`
+    );
+    selectOption = AppVersionSourceUpdateOption.SET_TO_LOCAL;
+    updateEasJson = false;
+  } else {
+    Log.log(
+      'Since EAS CLI version `11.0.0` explicitly specifying app version source is required. Please select your app version source:'
+    );
+    Log.log(
+      `\t1) With the "local" app version source and "autoIncrement" option enabled, the build number/version code is sourced from local project files and incremented automatically if possible, by editing local project files. ${learnMore(
+        'https://docs.expo.dev/build-reference/app-versions/#local-version-source'
+      )}`
+    );
+    Log.log(
+      `\t2) With the "remote" app version source and "autoIncrement" option enabled, the build number/version code is stored on EAS servers and updated every time you create a new build. Remote auto-incrementation won't edit the version in the local project files, but instead, the new version will be injected automatically during the build process. ${learnMore(
+        'https://docs.expo.dev/build-reference/app-versions/#remote-version-source'
+      )}`
+    );
+    Log.log(
+      `Until now, this project has been using the "local" version source (which was the previous default). App version source can now be set for you automatically, or you can configure it manually by setting the "appVersionSource" value in your eas.json.`
+    );
+
+    selectOption = await selectAsync(`What would you like to do?`, [
+      {
+        title: 'Update eas.json to use the default "remote" version source (recommended)',
+        value: AppVersionSourceUpdateOption.SET_TO_REMOTE,
+      },
+      {
+        title: 'Update eas.json to use "local" version source (old behavior)',
+        value: AppVersionSourceUpdateOption.SET_TO_LOCAL,
+      },
+      {
+        title: "Don't update eas.json, abort command and configure manually",
+        value: AppVersionSourceUpdateOption.ABORT,
+      },
+    ]);
+    updateEasJson = true;
+  }
+
+  if (selectOption === AppVersionSourceUpdateOption.SET_TO_LOCAL) {
+    if (updateEasJson) {
+      await easJsonAccessor.readRawJsonAsync();
+      easJsonAccessor.patch(easJsonRawObject => {
+        easJsonRawObject.cli = {
+          ...easJsonRawObject?.cli,
+          appVersionSource: AppVersionSource.LOCAL,
+        };
+        return easJsonRawObject;
+      });
+      await easJsonAccessor.writeAsync();
+    }
+    if (easJsonCliConfig) {
+      easJsonCliConfig.appVersionSource = AppVersionSource.LOCAL;
+    }
```

Comment:
awesome 👏 

did you test that `local` will be respected here for `--non-interactive` even if `eas.json` file wasn't updated?

Reply:
Pretty sure I did, but I will again

Reply:
Yes, it is